### PR TITLE
Refactors gas reagents

### PR DESCRIPTION
**Pull request by SapphicOverload**
# Document the changes in your pull request

Refactors how gas breath effects work, and the gas reagents themselves. Also fixes preterni sometimes not being affected by certain gases, and allows any gas with reagent effects to work in cryo tubes.

Closes #22187

# Why is this good for the game?
Significantly better code, allows most gases with breath effects to work in cryo tubes.

# Testing

image

# Changelog

:cl:  
experimental: Refactored gas reagents and gas breath effects
tweak: Most gases with breath effects now work in cryo tubes
bugfix: Fixed certain gases sometimes not having effects on preterni
/:cl:

### CODE REVIEW

**Comment by wonderinghost** [comment]
Gay

**Comment by ShoziOffical** [comment]
hot

**Comment by ynot01** [comment]
Solve lints

File: code/modules/atmospherics/machinery/components/unary_devices/cryo.dm [hunk 1]
```diff
@@ -228,6 +228,7 @@
 	. = on
 	on = new_value
 	update_appearance()
+	CRASH("AAAAAAAA")
```
Comment (ynot01) [inline code/modules/atmospherics/machinery/components/unary_devices/cryo.dm]: Remove debug message

**Review by ynot01 — CHANGES_REQUESTED** [review]

**Review by ynot01 — APPROVED** [review]

**Pull request merged**